### ISSUE TITLE
Provide a way to include warning messages in .jshintrc

### ISSUE THREAD

**Issue by Adam-Moss** [jshint/jshint#2575]
This is basically the opposite of #2551 in that I'd like to be able to include specific warnings in the .jshintrc.  For example if I have in my .jshintrc:

    "globalstrict": false,
    "strict": true,
    "node": true

Then the global strict warning W097 is effectively ignored.

I've tried:

    "globalstrict": false,
    "strict": true,
    "node": true,
    "predef": [
        "-console"
      ],
    "+W097": true

Which (based on #1922 and #2551) I believe should do what I'm looking to achieve but sadly doesn't.

**Comment by lukeapage** [comment]
I think this is a specific problem, not a general one. We will likely never have the ability to turn off an option and switch a warning on, because the options change the way jshint runs, which generate different warnings.

The problem is because of this line: https://github.com/jshint/jshint/blob/561d9785cdca88fc0a3178795274207520d05596/src/jshint.js#L5146

I think what you are looking for is more a way to enforce W097 in a node environment.

Why? is your environment both node and browser?

Have you considered switching off the node option and adding the globals you want to be available ?

**Comment by Adam-Moss** [comment]
Yes, I was looking to enforce W097 in a node environment, which was purely for convenience to ensure consistency between browser and node code.

Actually switching of the node option and adding the globals I wanted was the first thought I had but I prefer the semantics of having the environment definition (OCD asthetics :smile: ), and thought it would be easy enough to include the exceptions I wanted (in this case W097).

So I guess based on that `if` statement you highlighted what I'd like to be able to do is:

    if (!(state.option.module || state.option.node || state.option.phantom ||
              state.option.browserify) || <<w097_forcibly_switched_on>> )

If that makes sense?

**Comment by nicolo-ribaudo** [comment]
If #2567 lands, you will be able to use `strict: func` to enforce W097 in a node environment:
```js
/* jshint node: true, strict: func */

"use strict"; // Use the function form of "use strict".
```

**Comment by Adam-Moss** [comment]
@nicolo-ribaudo Perfect!  That is exactly what I'm looking for :grinning:  Fingers crossed that can be merged in. :+1:

**Comment by Adam-Moss** [comment]
Actually having thought a bit more about this PR I'm not sure it's the right way to go.  If I read the JSHint options documentation at http://jshint.com/docs/options/ there are quite a few of the options being deprecated as they are not linting issues but rather stylistic choices.

Following that thought train would therefore suggest the same should be applied to this requirement, as the current behaviour of jshint in this instance is to force a coding style in this scenario.

So the question would be therefore is the "correct" behaviour when specifying an environment in jshint for it to actually just define the globals associated with that environment rather than define the globals and some rules?  Because presumably the rules bit would be covered anyway if you set the relevant option?
